tools/metricsdocs/metricsdocs.go

```diff
@@ -84,6 +112,22 @@ func readFromPrometheusCR() metricList {
 	return ml
 }
 
+func getComments(reg *markers.Registry, ruleNode *yaml.Node) (string, string) {
+	if ruleNode.HeadComment == "" {
+		return "", ""
+	}
+
+	defn := reg.Lookup(ruleNode.HeadComment, markers.DescribesPackage)
+	rawComment, err := defn.Parse(ruleNode.HeadComment)
+	if err != nil {
+		return "", ""
+	}
+
+	comment := rawComment.(Comment)
```

Comment:
Can the assertion fail? Could you check that the type assertion succeeded? This function should return an error if the assertion doesn't succeed.

Reply:
check added